Panic on cilium-agent startup

## Bug report

**General Information**

- Cilium version (run `cilium version`)
  - `1.8.5`
- Kernel version (run `uname -a`)
  - `4.19.112+`
- Orchestration system version in use (e.g. `kubectl version`, Mesos, ...)
  - `Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.9-gke.6300", GitCommit:"eb6985a7ebfd53457b0b91ba08fac07597bb87af", GitTreeState:"clean", BuildDate:"2020-09-15T09:20:11Z", GoVersion:"go1.13.9b4", Compiler:"gc", Platform:"linux/amd64"}`
- Link to relevant artifacts (policies, deployments scripts, ...)

Trying to upgrade from `1.7.11`, previously this host had `1.8.4` too

```
level=info msg="Restored services from maps" failed=0 restored=73 subsys=service
panic: interface conversion: *lbmap.SourceRangeKey4 is not lbmap.SourceRangeKey: missing method ToHost

goroutine 1 [running]:
github.com/cilium/cilium/pkg/maps/lbmap.(*LBBPFMap).DumpSourceRanges.func1(0x285dfc0, 0xc000bb3c00, 0x283c100, 0xc000bb3c0c)
	/go/src/github.com/cilium/cilium/pkg/maps/lbmap/lbmap.go:235 +0x62
github.com/cilium/cilium/pkg/bpf.(*Map).DumpWithCallback(0xc0000e4d20, 0xc00081c610, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/bpf/map_linux.go:617 +0x61c
github.com/cilium/cilium/pkg/maps/lbmap.(*LBBPFMap).DumpSourceRanges(0x4116998, 0xc00081c600, 0x0, 0x0, 0x1)
	/go/src/github.com/cilium/cilium/pkg/maps/lbmap/lbmap.go:247 +0x72
github.com/cilium/cilium/pkg/service.(*Service).restoreAndDeleteOrphanSourceRanges(0xc000648e00, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/service/service.go:450 +0x234
github.com/cilium/cilium/pkg/service.(*Service).RestoreServices(0xc000648e00, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/service/service.go:384 +0x2a1
github.com/cilium/cilium/daemon/cmd.NewDaemon(0x285d500, 0xc0005ac900, 0xc0005acd40, 0x2896980, 0xc000648b60, 0xc000541040, 0x2896980, 0xc000648b60, 0xc00081e4f8)
	/go/src/github.com/cilium/cilium/daemon/cmd/daemon.go:393 +0x3bef
github.com/cilium/cilium/daemon/cmd.runDaemon()
	/go/src/github.com/cilium/cilium/daemon/cmd/daemon_main.go:1300 +0x414
github.com/cilium/cilium/daemon/cmd.glob..func1(0x3b9b1a0, 0xc000418020, 0x0, 0x1)
	/go/src/github.com/cilium/cilium/daemon/cmd/daemon_main.go:122 +0x13a
github.com/spf13/cobra.(*Command).execute(0x3b9b1a0, 0xc000116050, 0x1, 0x1, 0x3b9b1a0, 0xc000116050)
	/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x3b9b1a0, 0xc0001b2120, 0x44421a, 0x3b07a00)
	/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:887
github.com/cilium/cilium/daemon/cmd.Execute()
	/go/src/github.com/cilium/cilium/daemon/cmd/daemon_main.go:147 +0x5b
main.main()
	/go/src/github.com/cilium/cilium/daemon/main.go:22 +0x20
```

Reverting https://github.com/cilium/cilium/commit/aba794a382872d0d573b23483c1f12d22cc4ef1e fixes the issue

--- a/pkg/maps/lbmap/source_range.go
+++ b/pkg/maps/lbmap/source_range.go
@@ -55,7 +55,7 @@ type SourceRangeKey4 struct {
 func (k *SourceRangeKey4) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
 func (k *SourceRangeKey4) NewValue() bpf.MapValue    { return &SourceRangeValue{} }
 func (k *SourceRangeKey4) String() string            { return fmt.Sprintf("%s", k.Address) }
-func (k *SourceRangeKey4) ToNetwork() *SourceRangeKey4 {
+func (k *SourceRangeKey4) ToNetwork() SourceRangeKey {
 	n := *k
 	// For some reasons rev_nat_index is stored in network byte order in
 	// the SVC BPF maps
@@ -64,7 +64,7 @@ func (k *SourceRangeKey4) ToNetwork() *SourceRangeKey4 {
 }
 
 // ToHost returns the key in the host byte order
-func (k *SourceRangeKey4) ToHost() *SourceRangeKey4 {
+func (k *SourceRangeKey4) ToHost() SourceRangeKey {
 	h := *k
 	h.RevNATID = byteorder.NetworkToHost(h.RevNATID).(uint16)
 	return &h
@@ -96,7 +96,7 @@ type SourceRangeKey6 struct {
 func (k *SourceRangeKey6) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
 func (k *SourceRangeKey6) NewValue() bpf.MapValue    { return &SourceRangeValue{} }
 func (k *SourceRangeKey6) String() string            { return fmt.Sprintf("%s", k.Address) }
-func (k *SourceRangeKey6) ToNetwork() *SourceRangeKey6 {
+func (k *SourceRangeKey6) ToNetwork() SourceRangeKey {
 	n := *k
 	// For some reasons rev_nat_index is stored in network byte order in
 	// the SVC BPF maps
@@ -105,7 +105,7 @@ func (k *SourceRangeKey6) ToNetwork() *SourceRangeKey6 {
 }
 
 // ToHost returns the key in the host byte order
-func (k *SourceRangeKey6) ToHost() *SourceRangeKey6 {
+func (k *SourceRangeKey6) ToHost() SourceRangeKey {
 	h := *k
 	h.RevNATID = byteorder.NetworkToHost(h.RevNATID).(uint16)
 	return &h

